File: PFE_Nemoh/MyTestCases/Validation/Validation.py

from scipy.special import jv, yv
import numpy as np
PI=4.*np.arctan(1.)
N=4
print("N=", N)
X=1
H_1=jv(N, X)+1j*yv(N, X)
H_2=jv(N, X)-1j*yv(N, X)

# print("H_1", H_1)
print("H_2", H_2)

# Hankel_2 = -1j * jv(N, X) * (np.exp(PI*N*1j) -(-1)**N)/ np.sin(PI * N)
Hankel_2 = -1j * (jv(N, X) * np.exp(PI*N*1j)-jv(-N, X))/ np.sin(PI * N)
print("Hankel", Hankel_2)

# print("test", yv(N,X), (-1)**(N+1/2)*jv(-N,X))

File: PFE_Nemoh/MyTestCases/Validation/Validation.f90
PROGRAM Validation

    USE MBESSEL,            ONLY:fun_BESSJ

    IMPLICIT NONE

    REAL, PARAMETER           :: PI=4.*ATAN(1.)
    COMPLEX, PARAMETER        :: II=CMPLX(0.,1.)
    INTEGER :: N
    COMPLEX :: Hankel_2, EXP_term, correction
    REAL :: X

    N = 1
    X = 1.0

    EXP_term = CMPLX(COS(PI * N), SIN(PI * N))
    correction = EXP_term - CMPLX((-1)**N, 0.0)

    Hankel_2 = -II * fun_BESSJ(N, X) * correction / SIN(PI * N)

    print*, "Jn =", fun_BESSJ(N, X)
    print*, "EXP(II * PI * N) =", EXP_term
    print*, "Correction term =", correction
    print*, "Hn =", Hankel_2


END PROGRAM